### PR TITLE
[chore][processor/batchprocessor] Enable goleak check

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Add `goleak` to detect leaks in tests. Leaking goroutines were detected that were caused by a dependency that we can ignore (`go.opencensus.io/stats/view.(*worker).start`), some `Shutdown` calls were also added that were missing.

**Link to tracking Issue:** <Issue number if applicable>
#9165 

**Testing:** <Describe what testing was performed and which tests were added.>
Added check is passing as well as existing tests.

### CODE REVIEW

File: processor/batchprocessor/factory_test.go [hunk 1]
```diff
@@ -29,12 +29,15 @@ func TestCreateProcessor(t *testing.T) {
 	tp, err := factory.CreateTracesProcessor(context.Background(), creationSet, cfg, nil)
 	assert.NotNil(t, tp)
 	assert.NoError(t, err, "cannot create trace processor")
+	defer func() { assert.NoError(t, tp.Shutdown(context.Background())) }()
```
Comment (bogdandrutu) [inline processor/batchprocessor/factory_test.go:32]: Don't need to defer here, since we can shutdown here because you don't use the tp later.

**Review by bogdandrutu — APPROVED** [review]

**Comment by crobert-1** [comment]
Failing test is frequency of #9230

File: processor/batchprocessor/factory_test.go [hunk 2]
```diff
@@ -29,12 +29,15 @@ func TestCreateProcessor(t *testing.T) {
 	tp, err := factory.CreateTracesProcessor(context.Background(), creationSet, cfg, nil)
 	assert.NotNil(t, tp)
 	assert.NoError(t, err, "cannot create trace processor")
+	assert.NoError(t, tp.Shutdown(context.Background()))
 
 	mp, err := factory.CreateMetricsProcessor(context.Background(), creationSet, cfg, nil)
 	assert.NotNil(t, mp)
 	assert.NoError(t, err, "cannot create metric processor")
+	defer func() { assert.NoError(t, mp.Shutdown(context.Background())) }()
```
Comment (bogdandrutu) [inline processor/batchprocessor/factory_test.go:37]: same

File: processor/batchprocessor/factory_test.go [hunk 3]
```diff
@@ -29,12 +29,15 @@ func TestCreateProcessor(t *testing.T) {
...
 
 	lp, err := factory.CreateLogsProcessor(context.Background(), creationSet, cfg, nil)
 	assert.NotNil(t, lp)
 	assert.NoError(t, err, "cannot create logs processor")
+	defer func() { assert.NoError(t, lp.Shutdown(context.Background())) }()
```
Comment (bogdandrutu) [inline processor/batchprocessor/factory_test.go:42]: same

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**